heron/schedulers/src/java/com/twitter/heron/scheduler/ecs/EcsScheduler.java

```diff
@@ -0,0 +1,183 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.scheduler.ecs;
+
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import org.apache.commons.io.IOUtils;
+
+import com.twitter.heron.proto.scheduler.Scheduler;
+import com.twitter.heron.scheduler.utils.SchedulerUtils;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.scheduler.IScheduler;
+import com.twitter.heron.spi.utils.ShellUtils;
+
+
+/**
+ * Created by ananth on 4/19/17.
+ */
+public class EcsScheduler implements IScheduler {
+  private static final Logger LOG = Logger.getLogger(EcsScheduler.class.getName());
+  private Config config;
+  private Config runtime;
+
+  private volatile boolean isTopologyKilled = false;
+  private File tempDockerFile = null;
+  @Override
+  public void initialize(Config mConfig, Config mRuntime) {
+    this.config = Config.toClusterMode(mConfig);
+    this.runtime = mRuntime;
+
+  }
+
+  public void close() {
+
+  }
+
+
+  protected int startExecutorSyncProcess(int container) {
+    String executingInShell = new String();
+    executingInShell = getExecutorCommand(container)[0];
+    System.out.println("executing in Shell: " + executingInShell);
+    return ShellUtils.runProcess(executingInShell, null);
+  }
+
+  protected void startExecutor(final int container) {
+    LOG.info("Starting a new executor for container: " + container);
+    int shellOutput = startExecutorSyncProcess(container);
+    LOG.info("output value for the executor container: "
+        + container + String.valueOf(shellOutput));
+
+  }
+
+  private String[] getExecutorCommand(int container) {
+    List<Integer> freePorts = new ArrayList<>(SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR);
+    for (int i = 0; i < SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR; i++) {
+      //freePorts.add(SysUtils.getFreePort());
+      freePorts.add(5000 + (i + (container * 10)));
+    }
+    String[] executorCmd = SchedulerUtils.executorCommand(config, runtime, container, freePorts);
+    String finalExecCommand = setClusterValues(formHeronExecCommand(executorCmd));
+    String ecsTaskProject =  EcsContext.topologyName(config) + "_" + String.valueOf(container);
+    FileOutputStream dockerFilestream = null;
+    String content = null;
+    try {
+      tempDockerFile = File.createTempFile("docker", ".yml");
```

Comment:
The Default ECS Cluster Formation uses the ecs tasks where in we can pass the docker image and an entry point to start each of the containers. Not sure on the AMIs ..